apps/meteor/app/api/server/v1/rooms.ts

```diff
@@ -264,6 +264,10 @@ API.v1.addRoute(
 				return API.v1.failure('Body parameter "oldest" is required.');
 			}
 
+			if (limit && limit > settings.get<number>('Prune_message_limit')) {
```

Comment:
This should be the upper limit, if the user provides a limit greater than the setting it should be blocked. On the other hand, if the user doesn't pass any limit we should use the setting limit, unless it's not defined, in this case, we can delete all messages.